In [1]:
import pandas as pd


from ragas import evaluate
from datasets import Dataset    
from ragas.llms import LangchainLLMWrapper
from langchain_openai import ChatOpenAI
from dotenv import load_dotenv

import os

load_dotenv()
os.environ.get("OPENAI_API_KEY")

'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'

In [2]:
df = pd.read_json('./Llama_32_Instruct_k1.json')

# Rename columns to match required format
df = df.rename(columns={
    'query': 'user_input',
    'retrieved_chunk': 'retrieved_contexts', 
    'generated_response': 'response'
})

# Convert retrieved_chunk to array
df['retrieved_contexts'] = df['retrieved_contexts'].apply(lambda x: [x])

print(df.columns)

Index(['user_input', 'retrieved_contexts', 'response'], dtype='object')


In [3]:
# df.drop(columns=["thought"], inplace=True)
# print(df.columns)

from ragas.metrics import (
    faithfulness,
    answer_relevancy,
    context_precision,
    context_recall,
)

evaluator_llm = LangchainLLMWrapper(ChatOpenAI(model="gpt-4o-mini", temperature=0.5))

dataset_ragas = Dataset.from_pandas(df)
print(dataset_ragas)
result = evaluate(
    dataset_ragas,
    metrics=[
        faithfulness,
        answer_relevancy
    ],
    llm = evaluator_llm
)

print(result)

Dataset({
    features: ['user_input', 'retrieved_contexts', 'response'],
    num_rows: 194
})


Evaluating:   0%|          | 0/388 [00:00<?, ?it/s]

{'faithfulness': 0.5402, 'answer_relevancy': 0.4206}


llama 2 k1 {'faithfulness': 0.4132, 'answer_relevancy': 0.7586}  
llama 2 k3 {'faithfulness': 0.5930, 'answer_relevancy': 0.6805}  
llama 3 k1 {'faithfulness': 0.3255, 'answer_relevancy': 0.7992}  
llama 3 k3 {'faithfulness': 0.4729, 'answer_relevancy': 0.8132}   
llama 32 instruct k1 {'faithfulness': 0.5402, 'answer_relevancy': 0.4206}  

llama 2 k3 {'faithfulness': 0.5930, 'answer_relevancy': 0.6805}

llama 3 k1 {'faithfulness': 0.3255, 'answer_relevancy': 0.7992}

llama 3 k3 {'faithfulness': 0.4729, 'answer_relevancy': 0.8132}


llama 32 instruct k1 {'faithfulness': 0.5402, 'answer_relevancy': 0.4206}